### ISSUE TITLE
Deployment failure when using binary modules.

### ISSUE THREAD

**Issue by Rokt33r** [vercel/vercel#639]
I'm using `sharp` to process images. Is there anything I can do?

The following log is from now.sh.

```
06/06 09:05 PM (5m)
npm install
06/06 09:05 PM (5m)
✓ Using "yarn.lock"
06/06 09:05 PM (5m)
⧗ Installing 44 main dependencies…
06/06 09:06 PM (4m)
> Building "sharp@0.18.1" remotely

> sharp@0.18.1 install /tmp/ace-build-sharp-0.18.1-g4lpI/node_modules/sharp
> node-gyp rebuild
06/06 09:06 PM (4m)
ERROR: musl libc (x86_64) requires manual installation - please see http://sharp.dimens.io/en/stable/install/
gyp: Call to 'LDD_VERSION="musl libc (x86_64)
Version 1.1.16
Dynamic Program Loader
Usage: ldd [options] [--] pathname" node -e "require('./binding').download_vips()"' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
06/06 09:06 PM (4m)
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/node/57/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:336:16)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:197:12)
gyp ERR! System Linux 4.7.0-coreos
gyp ERR! command "/node/57/bin/node" "/node/57/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /tmp/ace-build-sharp-0.18.1-g4lpI/node_modules/sharp
06/06 09:06 PM (4m)
gyp ERR! node -v v8.0.0
gyp ERR! node-gyp -v v3.6.1
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sharp@0.18.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
06/06 09:06 PM (4m)
npm ERR! Failed at the sharp@0.18.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/nowuser/.npm/_logs/2017-06-06T12_06_10_912Z-debug.log
06/06 09:06 PM (4m)
> Building "sharp@0.18.1" remotely

> sharp@0.18.1 install /tmp/ace-build-sharp-0.18.1-WsbS7/node_modules/sharp
> node-gyp rebuild
06/06 09:06 PM (4m)
ERROR: musl libc (x86_64) requires manual installation - please see http://sharp.dimens.io/en/stable/install/
gyp: Call to 'LDD_VERSION="musl libc (x86_64)
Version 1.1.16
Dynamic Program Loader
Usage: ldd [options] [--] pathname" node -e "require('./binding').download_vips()"' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
06/06 09:06 PM (4m)
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/node/57/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:336:16)
gyp ERR! stack     at emitTwo (events.js:125:13)
06/06 09:06 PM (4m)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:197:12)
gyp ERR! System Linux 4.7.0-coreos
06/06 09:06 PM (4m)
gyp ERR! command "/node/57/bin/node" "/node/57/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /tmp/ace-build-sharp-0.18.1-WsbS7/node_modules/sharp
gyp ERR! node -v v8.0.0
gyp ERR! node-gyp -v v3.6.1
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sharp@0.18.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sharp@0.18.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
06/06 09:06 PM (4m)
npm ERR!     /home/nowuser/.npm/_logs/2017-06-06T12_06_16_933Z-debug.log
06/06 09:06 PM (4m)
> Building "sharp@0.18.1" remotely

> sharp@0.18.1 install /tmp/ace-build-sharp-0.18.1-yNVes/node_modules/sharp
> node-gyp rebuild
06/06 09:06 PM (4m)
ERROR: musl libc (x86_64) requires manual installation - please see http://sharp.dimens.io/en/stable/install/
gyp: Call to 'LDD_VERSION="musl libc (x86_64)
Version 1.1.16
Dynamic Program Loader
Usage: ldd [options] [--] pathname" node -e "require('./binding').download_vips()"' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/node/57/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:336:16)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:197:12)
gyp ERR! System Linux 4.7.0-coreos
gyp ERR! command "/node/57/bin/node" "/node/57/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /tmp/ace-build-sharp-0.18.1-yNVes/node_modules/sharp
gyp ERR! node -v v8.0.0
gyp ERR! node-gyp -v v3.6.1
gyp ERR! not ok 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sharp@0.18.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sharp@0.18.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/nowuser/.npm/_logs/2017-06-06T12_06_22_873Z-debug.log
06/06 09:06 PM (4m)
> Build failed for "sharp@0.18.1" on Linux/musl 64-bit with Unsupported runtime (57): exit code 1

Error: Error parsing `package.json` for sharp-0.18.1.tar
    at extract (/snapshot/home/travis/build/zeit/ace/lib/extract.js:36:11)
    at process._tickCallback (internal/process/next_tick.js:103:7)
Error: Error parsing `package.json` for sharp-0.18.1.tar
    at extract (/snapshot/home/travis/build/zeit/ace/lib/extract.js:36:11)
    at process._tickCallback (internal/process/next_tick.js:103:7)
Error: Error parsing `package.json` for sharp-0.18.1.tar
    at extract (/snapshot/home/travis/build/zeit/ace/lib/extract.js:36:11)
    at process._tickCallback (internal/process/next_tick.js:103:7)
06/06 09:06 PM (4m)
npm install
06/06 09:06 PM (4m)
Installing package aws-sdk@^2.54.0
06/06 09:06 PM (4m)
Installing package body-parser@~1.17.1
06/06 09:06 PM (4m)
Installing package cookie-parser@~1.4.3
06/06 09:06 PM (4m)
Installing package debug@~2.6.3
06/06 09:06 PM (4m)
Installing package dotenv@^4.0.0
06/06 09:06 PM (4m)
Installing package express@~4.15.2
06/06 09:06 PM (4m)
Installing package express-fileupload@^0.1.3
06/06 09:06 PM (4m)
Installing package express-session@^1.15.2
06/06 09:06 PM (4m)
Installing package http-proxy-middleware@^0.17.4
06/06 09:06 PM (4m)
Installing package moment@^2.18.1
06/06 09:06 PM (4m)
Installing package mongoose@^4.9.4
06/06 09:06 PM (4m)
Installing package morgan@~1.8.1
06/06 09:06 PM (4m)
Installing package next@^2.4.1
06/06 09:06 PM (4m)
Installing package nprogress@^0.2.0
06/06 09:06 PM (4m)
Installing package octicons@^5.0.1
06/06 09:06 PM (4m)
Installing package open-color@^1.5.1
06/06 09:06 PM (4m)
Installing package passport@^0.3.2
06/06 09:06 PM (4m)
Installing package passport-github@^1.1.0
06/06 09:06 PM (4m)
Installing package react@^15.5.3
06/06 09:06 PM (4m)
Installing package react-dom@^15.5.3
06/06 09:06 PM (4m)
Installing package react-redux@^5.0.4
Installing package redux@^3.6.0
06/06 09:06 PM (4m)
Installing package redux-logger@^3.0.1
06/06 09:06 PM (4m)
Installing package redux-saga@^0.15.3
06/06 09:06 PM (4m)
Installing package reduxlet-saga@^0.3.0
06/06 09:06 PM (4m)
Installing package remark@^7.0.1
06/06 09:06 PM (4m)
Installing package remark-react@^4.0.0
06/06 09:06 PM (4m)
Installing package session-file-store@^1.0.0
Installing package sharp@^0.18.1
Installing package socket.io@^1.7.4
06/06 09:06 PM (4m)
Installing package twemoji@^2.2.5
06/06 09:06 PM (4m)
Installing package validator@^7.0.0
06/06 09:06 PM (4m)
Installing package @carbonstack/base-css@^0.0.3
06/06 09:06 PM (4m)
Installing package async-to-gen@^1.3.3
06/06 09:06 PM (4m)
Installing package babel-eslint@^7.2.2
06/06 09:06 PM (4m)
Installing package babel-jest@^19.0.0
06/06 09:06 PM (4m)
Installing package babel-plugin-inline-react-svg@^0.4.0
06/06 09:06 PM (4m)
Installing package babel-plugin-transform-define@^1.2.0
06/06 09:06 PM (4m)
Installing package babel-plugin-transform-runtime@^6.23.0
06/06 09:06 PM (4m)
Installing package babel-preset-stage-0@^6.24.1
06/06 09:06 PM (4m)
Installing package concurrently@^3.4.0
06/06 09:06 PM (4m)
Installing package jest@^19.0.2
06/06 09:06 PM (4m)
Installing package nodemon@^1.11.0
06/06 09:06 PM (4m)
Installing package standard@^10.0.1
06/06 09:07 PM (3m)
(node:83) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: ENOENT: no such file or directory, readlink '/home/nowuser/src/node_modules/.tmp/fe83890a-2798-4488-a9f4-21f6e869ec6f/node_modules/node-pre-gyp'
(node:83) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
06/06 09:07 PM (3m)
sh: node-gyp: not found
06/06 09:07 PM (3m)
Error: Command node-gyp rebuild ended with code 127
    at ChildProcess.<anonymous> (/usr/lib/node_modules/@zeit/cace/build/lib/run.js:1:599)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at maybeClose (internal/child_process.js:887:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:208:5)
06/06 09:07 PM (3m)
npm run build
06/06 09:07 PM (3m)
> carbonstack@0.1.0 build /home/nowuser/src
> next build
06/06 09:07 PM (3m)
> Using external babel configuration
06/06 09:07 PM (3m)
> location: "/home/nowuser/src/.babelrc"
06/06 09:08 PM (2m)
npm start
06/06 09:08 PM (2m)
> carbonstack@0.1.0 start /home/nowuser/src
> NODE_ENV=production node next/app.js
06/06 09:08 PM (2m)
(node:250) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Cannot find module '../build/Release/sharp.node'
06/06 09:08 PM (2m)
(node:250) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
06/06 09:08 PM (2m)
Db.prototype.authenticate method will no longer be available in the next major release 3.x as MongoDB 3.6 will only allow auth against users in the admin db and will no longer allow multiple credentials on a socket. Please authenticate using MongoClient.connect with auth credentials.
06/06 09:08 PM (2m)

```

**Comment by leo** [comment]
@TooTallNate

**Comment by TooTallNate** [comment]
@Rokt33r please use `sharp@0.17.3` for now. The latest version of sharp requires a version of libvips that is not yet available to out build nodes.

**Comment by TooTallNate** [comment]
Once apk updates their vips to >= 8.5.5 then the latest version should work:  https://pkgs.alpinelinux.org/package/edge/testing/x86_64/vips

**Comment by Rokt33r** [comment]
Thank you so much! The problem is fixed after I installed sharp@0.17.3.

**Comment by TooTallNate** [comment]
And now the latest version of `sharp` works again. Thank you for your patience @Rokt33r!
